src/algebra/module.lean

```diff
@@ -411,7 +443,7 @@ by refine {to_fun := coe, ..}; simp [coe_smul]
 
 @[simp] theorem subtype_apply (x : p) : p.subtype x = x := rfl
 
-lemma subtype_eq_val (p : submodule R M) :
+lemma subtype_eq_val :
   ((submodule.subtype p) : p → M) = subtype.val := rfl
```

Comment:
```suggestion
lemma subtype_eq_val : ((submodule.subtype p) : p → M) = subtype.val := rfl
```

Reply:
Should this be `norm_cast` and/or `simp`?

Reply:
I'm not sure it should be. As soon as you `simp` this away, you lose the ability to use lemmas about `R`-linearity.

Reply:
Agreed.